crates/store/re_chunk_store/src/dataframe.rs

```diff
@@ -398,6 +398,153 @@ impl ComponentColumnDescriptor {
     }
 }
 
+// --- Selectors ---
+
+/// Describes a column selection to return as part of a query.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum ColumnSelector {
+    Control(ControlColumnSelector),
+    Time(TimeColumnSelector),
+    Component(ComponentColumnSelector),
+    //TODO(jleibs): Add support for archetype-based component selection.
+    //ArchetypeField(ArchetypeFieldColumnSelector),
+}
+
+impl From<ColumnDescriptor> for ColumnSelector {
+    fn from(desc: ColumnDescriptor) -> Self {
+        match desc {
+            ColumnDescriptor::Control(desc) => Self::Control(desc.into()),
+            ColumnDescriptor::Time(desc) => Self::Time(desc.into()),
+            ColumnDescriptor::Component(desc) => Self::Component(desc.into()),
+        }
+    }
+}
+
+impl From<ControlColumnSelector> for ColumnSelector {
+    fn from(desc: ControlColumnSelector) -> Self {
+        Self::Control(desc)
+    }
+}
+
+impl From<TimeColumnSelector> for ColumnSelector {
+    fn from(desc: TimeColumnSelector) -> Self {
+        Self::Time(desc)
+    }
+}
+
+impl From<ComponentColumnSelector> for ColumnSelector {
+    fn from(desc: ComponentColumnSelector) -> Self {
+        Self::Component(desc)
+    }
+}
+
+/// Select a control column.
+///
+/// The only control column currently supported is `rerun.components.RowId`.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ControlColumnSelector {
+    /// Name of the control column.
+    pub component: ComponentName,
+}
+
+impl From<ControlColumnDescriptor> for ControlColumnSelector {
+    fn from(desc: ControlColumnDescriptor) -> Self {
+        Self {
+            component: desc.component_name,
+        }
+    }
+}
+
+/// Select a time column.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct TimeColumnSelector {
+    /// The name of the timeline.
+    pub timeline: TimelineName,
```

Comment:
I like that these `*Selector` have the simplest possible types. Having extact 1:1 fbs counterparts will make everyone's life easy 👍🏻